### PR TITLE
Balenalib2

### PR DESCRIPTION
**Pull request by dansku**
Renaming from resin to balena

### CODE REVIEW

**Comment by lurch** [comment]
I guess this fixes #84 ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file
 automatically by Versionist. DO NOT EDIT THIS FILE MANUALLY!
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+# v.0.1.1
```
Comment (imrehg) [inline CHANGELOG.md]: Why is this edit in here? It should not be edited manually, as it is created by versionbot.

**Comment by imrehg** [comment]
@dansku where did you change things to `balenalib`? I don't see anything like that in the changed files, and that was already done in #88 ?

**Comment by dansku** [comment]
@imrehg yeah I noticed that it was changed and not merged after I updated the files (if I remember correctly). But I also updated the readme and the files.

File: CHANGELOG.md (hunk 1 above)
Comment (dansku) [reply CHANGELOG.md]: Agreed!

File: README.md [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
 
 ## Getting started
 
-- Sign up on [resin.io](https://dashboard.resin.io/signup)
-- go throught the [getting started guide](http://docs.resin.io/raspberrypi/nodejs/getting-started/) and create a new application
-- clone this repository to your local workspace
-- add the _resin remote_ to your local workspace using the useful shortcut in the dashboard UI ![remoteadd](https://raw.githubusercontent.com/resin-io-playground/boombeastic/master/docs/gitresinremote.png)
-- `git push resin master`
+- Sign up on [balena](https://dashboard.balena-cloud.com/signup)
+- go throught the [getting started guide](https://www.balena.io/docs/learn/getting-started/raspberry-pi/nodejs/) and create a new application
```
Comment (lurch) [inline README.md]: throught -> through

File: app/main.js [hunk 3]
```diff
@@ -66,29 +66,29 @@ if (process.env.NODE_ENV === 'development') {
...
-if (electronConfig.ELECTRON_RESIN_UPDATE_LOCK) {
+if (electronConfig.ELECTRON_BALENA_UPDATE_LOCK) {
   const lockFile = require('lockfile');
-  electron.ipcMain.on('resin-update-lock', (event, command) => {
+  electron.ipcMain.on('balena-update-lock', (event, command) => {
     switch (command) {
       case 'lock':
-        lockFile.lock('/tmp/resin/resin-updates.lock', (error) => {
-          event.sender.send('resin-update-lock', error);
+        lockFile.lock('/tmp/balena/balena-updates.lock', (error) => {
```
Comment (lurch) [inline app/main.js]: According to https://www.balena.io/docs/learn/deploy/release-strategy/update-locking/ the lockfile path should be `/tmp/balena/updates.lock` ? :eyes:

File: Dockerfile.template [hunk 4]
```diff
@@ -62,13 +68,92 @@ RUN install_packages \
...
+CMD ["bash", "/usr/src/app/start.sh"]###
+# Build stage
+##
+FROM balenalib/%%BALENA_MACHINE_NAME%%-node:10-build as build
+
+# Move to app dir
+WORKDIR /usr/src/app
+
+# Install build dependencies
+RUN install_packages \
```
Comment (dansku) [inline Dockerfile.template]: Oh, vim went crazy here. Fixing...

**Comment by dansku** [comment]
My bad, just saw that I was on a wrong local repo as I was editing these files a while ago. Pushing the fixes now.

File: app/package.json [hunk 5]
```diff
@@ -27,9 +27,9 @@
   "author": "Carlo Maria Curinga",
   "license": "Apache-2.0",
   "bugs": {
-    "url": "https://github.com/resin-io/resin-electronjs/issues"
+    "url": "https://github.com/balena-io/balena-electronjs/issues"
```
Comment (lurch) [inline app/package.json:30]: This URL doesn't work yet - I guess somebody with the necessary privileges will need to rename this repo after this PR is merged?

Comment (dansku) [reply app/package.json:30]: Yeah, we could ask @shaunmulligan to do it I believe

Comment (shaunmulligan) [reply app/package.json:30]: @dansku yeah can update it once merged, if you give me a ping.

Comment (dansku) [reply app/package.json:30]: @shaunmulligan will do!

**Review by imrehg — APPROVED** [review]
LGTM, let's get this over with, sorry for the delay, @dansku

File: app/package.json (hunk 5 above)
Comment (lurch) [reply app/package.json:30]: ping @shaunmulligan I guess it's time for this repo to be renamed? (as this PR has been merged)

File: app/main.js [hunk 6]
```diff
@@ -66,29 +66,29 @@ if (process.env.NODE_ENV === 'development') {
...
 }
 
-// Listen for a 'resin-update-lock' to either enable, disable or check
+// Listen for a 'update-lock' to either enable, disable or check
 // the update lock from the renderer process (i.e. the app)
-if (electronConfig.ELECTRON_RESIN_UPDATE_LOCK) {
+if (electronConfig.ELECTRON_BALENA_UPDATE_LOCK) {
   const lockFile = require('lockfile');
-  electron.ipcMain.on('resin-update-lock', (event, command) => {
+  electron.ipcMain.on('update-lock', (event, command) => {
```
Comment (lurch) [inline app/main.js:73]: I've just noticed that the example code in `README.md` is using e.g. `ipcRenderer.once('balena-update-lock', (event, error) => {`
Does it need to be updated to instead use the `'update-lock'` name that is being used in `main.js` ?

**Pull request merged**